### PR TITLE
fix(web): restore endorsed skills grid + pin links to canonical GitHub URL

### PR DESCRIPTION
**Pull request by BA-CalderonMorales**
## Problem

The **Endorsed Skills** section on `/skills` (`apps/web`) is broken: it renders empty in production and, when populated, its links can resolve to a 404.

The page fetches the endorsed skill list from the `solana-dev-skill` repo via the GitHub contents API. That repo was restructured — the reference markdown now lives under the canonical `skills/solana-dev/references/` (plural) path — but the app still requested the stale singular `skill/references` path, which now returns `404`. The fetch fails silently, so the endorsed grid shows nothing.

Separately, the parser that builds each endorsed link did not defensively pin the link target to the canonical GitHub `html_url`. If a skill's frontmatter ever carried a stale/invalid `url` (e.g. a `/skill/...` singular path), that value could be used as the link target and send users to a 404 or the wrong page.

This matters because these links are the entry point to the Foundation-maintained skill content; unreliable targets erode trust in the page.

## Summary of Changes

Surgical fix in `apps/web`, scoped to the skills page data path:

1. **Correct the stale GitHub contents API path** — `apps/web/src/components/skills/skills.ts`:  `.../contents/skill/references` → `.../contents/skills/solana-dev/references`. This is the actual `/skill/` → `/skills/` bug; the endorsed grid loads again.

2. **Harden the endorsed link target** — `apps/web/src/app/[locale]/skills/skills.tsx`: `parseSkillMarkdown` now resolves the link via `resolveEndorsedSkillUrl`, which prefers the canonical GitHub `html_url` from the contents API and only honors a frontmatter `url` when its **parsed `URL().pathname`** matches the canonical `solana-dev-skill` reference blob path. Stale/invalid metadata — including a canonical segment smuggled into a query string or fragment, or a malformed URL — falls back to `html_url`. So endorsed links never point at a 404 or the wrong page.

3. **Regression tests** (narrowly scoped, Vitest):
   - `parseSkillMarkdown.test.ts` — stale `/skill/...` frontmatter → canonical fallback; no-frontmatter-url → canonical; query-string spoof rejected; fragment spoof rejected; a genuinely canonical frontmatter url is honored.
   - `skillsApiUrl.test.ts` — asserts the canonical API path and guards against the stale `/contents/skill/references` path.

No refactors or unrelated edits. Community skills and other page behavior are
untouched.

**Where the links now lead (verified locally):** each endorsed card opens `https://github.com/solana-foundation/solana-dev-skill/blob/main/skills/solana-dev/references/<name>.md` — the real reference content, not a 404.

## Local verification performed

Run from `apps/web` (Vitest is the test runner per `apps/web/AGENTS.md`):

- `pnpm --filter solana-com test -- src/__tests__/skills` — **21 files pass, 140 tests (18 skipped)**, including the 5 new URL tests + 2 API-path tests.
- `pnpm --filter solana-com lint` — **0 errors** (only pre-existing warnings in unrelated files).
- Live contents-API probe confirms all 11 reference files resolve under `/skills/solana-dev/references/`:

<details open>
<summary>Reproduce the canonical-path check</summary>

```js

const api =
  "https://api.github.com/repos/solana-foundation/solana-dev-skill/contents/skills/solana-dev/references";

const files = await fetch(api).then((res) => res.json());

files
  .filter((f) => f.name.endsWith(".md") && f.name !== "SKILL.md")
  .forEach((f) =>
    console.log(
      f.html_url.includes("/skills/solana-dev/references/") ? "OK " : "BAD",
      f.name,
    ),
);

```
</details>

- `pnpm --filter solana-com dev` → `http://localhost:3000/skills`: the Endorsed Skills grid renders cards (previously empty), and each card navigates to the correct canonical GitHub reference page. Confirmed manually.

> A minor finding from a local CodeRabbit CLI review (query/fragment segment
> smuggling in the URL guard) was addressed in its own commit
> (`fix(web): validate endorsed skill url pathname (CodeRabbit CLI)`), with the
> parsed-`pathname` validation and two added regression tests above.

## Fixes

**There doesn't appear to be an Open Issue about this, yet. Might be wrong here.**

## Change classification (SDLC §2)

- [ ] **Critical** — auth, secrets/key handling, fund movement, deploy/CI security gates, or anything that changes who can do what
- [x] **Standard** — production-facing, non-critical (features, API/route changes, dependency updates, monitoring/config)
- [ ] **Low** — no security impact (docs, tests, dev tooling, content)

## Security checklist

- [x] No secrets, tokens, or credentials in source, env files, or CI logs (use Doppler / `NEXT_PUBLIC_*` only for values safe to ship to the browser).
- [x] Input from users or external services is validated before use; no `dangerouslySetInnerHTML` / unsanitized HTML on untrusted input. (The GitHub-sourced frontmatter `url` is now validated via `URL().pathname` against a canonical allowlist before it can become a link target.)
- [x] New or updated dependencies are justified below, and `pnpm audit` passes (no new High/Critical advisories).
- [x] Errors are handled explicitly — no silent failures on production paths. (Malformed frontmatter URLs are caught and fall back to the canonical `html_url`; the existing fetch error path is unchanged.)
- [ ] For **Critical** changes: a trust-boundary / threat-model note is included below, and a second reviewer has been requested.

## New dependencies (name + why), or "none":

None

## Threat-model note for Critical changes, or "n/a":

N/A

Suggested label: `bug` (for maintainers to apply).

### CODE REVIEW

**Comment by BA-CalderonMorales** [comment]
Just an FYI to the maintainers here, I used OpenCode to help out here with finding the root cause. 

Found it when I was trying to create a trading bot and noticed the official skills weren't mapping to the right github skills page on the repo itself. 

Feel free to dm me on LinkedIn if you've got more questions or want to talk about any part of this PR. Will make sure to keep an eye on the PR itself, but hopefully this at least provides a starting point to patch this.

LinkedIn: https://www.linkedin.com/in/bcalderonmorales-cmoe/

Also, couldn't find an open issue that this patches, so left "Fixes #" intentionally blank. I also couldn't find anything explicitly stating we couldn't use coding agents to help with things like this. 

If that's true, I'll make sure to take this down to not cause strain on maintainers. 

Again, hope this helps!

File: apps/web/src/app/[locale]/skills/skills.tsx [hunk 1]
```diff
@@ -9,7 +9,39 @@ import { Divider } from "@/components/solutions/divider.v2";
 import { COMMUNITY_SKILLS } from "@/data/skills/communitySkills";
 import { SOLANA_DEV_SKILLS_GITHUB_API_URL } from "@/components/skills/skills";
 
-function parseSkillMarkdown(
+const CANONICAL_SKILLS_URL_PREFIX =
+  "https://github.com/solana-foundation/solana-dev-skill/blob/";
+const CANONICAL_SKILLS_PATH_SEGMENT = "/skills/solana-dev/references/";
+
+function resolveEndorsedSkillUrl(
```
Comment (BA-CalderonMorales) [inline apps/web/src/app/[locale]/skills/skills.tsx:16]: left this intentionally obvious in case you guys want to place this in some utils/ folder you guys have. 

would be open to moving it and then importing into this file if it's easier to maintain that way. 

lmk. just wanted to make sure something was in there so you guys can have it to shape it how you want down the line.

made sure to include test cases that reinforce the behavior that's wanted here with pointing folks to the skills repo with the official skills.

**Comment by BA-CalderonMorales** [comment]
To reproduce this, navigate to:

1. https://solana.com/skills
2. Try to click on one of the official skills.
3. Notice how it leads to a 404. 

After patch, this should point folks to where you want to get them to, along with patching root cause in some hard coded links that I found (w/ help of OpenCode).

File: apps/web/src/app/[locale]/skills/skills.tsx [hunk 2]
```diff
@@ -9,7 +9,39 @@ import { Divider } from "@/components/solutions/divider.v2";
...
+  // into a query/fragment) falls back to `html_url` so links never point at
+  // a 404 or the wrong page.
+  if (
+    typeof frontmatterUrl === "string" &&
+    frontmatterUrl.startsWith(CANONICAL_SKILLS_URL_PREFIX)
+  ) {
+    try {
+      if (
+        new URL(frontmatterUrl).pathname.includes(CANONICAL_SKILLS_PATH_SEGMENT)
+      ) {
```
[the remark this answers is not in the document]
Comment (BA-CalderonMorales) [reply apps/web/src/app/[locale]/skills/skills.tsx:35]: Solid call out from greptile. Tried to patch it with: https://github.com/solana-foundation/solana-com/pull/1720/commits/335126407c9c3f5033a1babe2e66c1b49712c9c5

Included some red/green tests to make sure edge cases mentioned are caught.

**Review by h4rkl — APPROVED** [review]

**Comment by h4rkl** [comment]
Awesome - thorough PR - much appreciated @BA-CalderonMorales

**Comment by BA-CalderonMorales** [comment]
@h4rkl made sure to boost Solana and the skills site with my connections on LinkedIn. 

Hope you guys don't mind the free marketing. 

https://www.linkedin.com/feed/update/urn:li:activity:7484810085379239936/

lmk if you'd rather I take the post down. Also using it to help out devs out there with an example of how to merge into codebases like this one.
